crates/sui-framework/packages/dwallet-system/sources/dwallet_transfer.move

```diff
@@ -0,0 +1,76 @@
+// Copyright (c) dWallet Labs, Ltd.
+// SPDX-License-Identifier: BSD-3-Clause-Clear
+
+#[allow(unused_use)]
+module dwallet_system::dwallet_transfer {
+    use dwallet::object::{Self, ID, UID};
+    use dwallet::token::recipient;
+    use dwallet::transfer;
+    use dwallet::tx_context;
+    use dwallet::tx_context::TxContext;
+    use dwallet_system::dwallet_2pc_mpc_ecdsa_k1::{DWallet, output};
+
+    const DWALLET_TRANSFER_ERROR: u64 = 0x1;
+
+    struct DwalletTransfer has key{
+        id: UID,
+        dwallet_id: ID,
+        encrypted_secret_share: vector<u8>,
+    }
+
+    struct PublicKey has key {
+        id: UID,
+        public_key: vector<u8>,
+        key_owner_address: address,
+    }
+
+    public fun store_public_key(key: vector<u8>, ctx: &mut TxContext): ID {
+        let pk = PublicKey {
+            id: object::new(ctx),
+            public_key: key,
+            key_owner_address: tx_context::sender(ctx),
+        };
+        let pk_id = object::id(&pk);
+        transfer::freeze_object(pk);
+        pk_id
+    }
+
+    public fun transfer_dwallet(
```

Comment:
So this is not transferring the dWallet, as the dWallet capability isn't even an argument of this function.
This function grants access to a new user for a dWallet.

In any case, @omersadika we should think from the user-experience perspective what kind of guarantee we want here.